### PR TITLE
Modernize clothing stores

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Overhaul clothing stores"

#### Purpose of change
Clothing stores feel like they're from 0.A. and they don't use modern mapgen.

#### Describe the solution
Update clothing stores to use palettes and SUS itemgroups. Add new multi z level clothing store maps.

Depends on #35932.

#### Testing
Game loads without errors and `s_clothes` is generated as expected.

### CODE REVIEW

**Comment by Moonsabrt** [comment]
Can you give us a demonstration of what they will look like now?  Or what you have planned for the extra story?

**Comment by ymber** [comment]
This should be ready now. I've reduced the scope because it was getting big and full of tangential changes. As it stands every `s_clothes*` map is updated except `s_clothes_3` and `s_clothes_5`. No new maps have been added and the updates to existing ones are mostly confined to items.

File: data/json/furniture_and_terrain/furniture-storage.json [hunk 1]
```diff
@@ -433,6 +433,32 @@
...
+    "bash": {
+      "str_min": 6,
+      "str_max": 30,
+      "sound": "metal screeching!",
+      "sound_fail": "clang!",
+      "items": [
+        { "item": "scrap", "count": [ 2, 8 ] },
+        { "item": "steel_chunk", "count": [ 2, 4 ] },
+        { "item": "sheet_metal_small", "count": [ 6, 10 ] },
+        { "item": "pipe", "count": 1 }
```
Comment (I-am-Erk) [inline data/json/furniture_and_terrain/furniture-storage.json]:
```suggestion
        { "item": "pipe", "count": 1 },
        { "item": "wheel_casters", "count": [ 0, 1 ] }
```

File: data/json/furniture_and_terrain/furniture-storage.json [hunk 2]
```diff
@@ -433,6 +433,32 @@
...
+    "id": "f_clothing_rail",
+    "name": "clothing rail",
+    "description": "A rail for hanging clothes on.",
+    "symbol": "{",
+    "color": "light_gray",
+    "move_cost_mod": -1,
+    "coverage": 10,
+    "required_str": 8,
+    "flags": [ "TRANSPARENT", "FLAMMABLE_HARD", "PLACE_ITEM", "BLOCKSDOOR" ],
+    "deconstruct": { "items": [ { "item": "pipe", "count": [ 4, 6 ] }, { "item": "sheet_metal_small", "count": [ 0, 4 ] } ] },
```
Comment (I-am-Erk) [inline data/json/furniture_and_terrain/furniture-storage.json]:
```suggestion
    "deconstruct": { "items": [ { "item": "pipe", "count": [ 4, 6 ] }, { "item": "sheet_metal_small", "count": [ 0, 4 ] } ] }, { "item": "wheel_casters", "count": 1 }
```

File: data/json/itemgroups/SUS/office.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  {
+    "id": "SUS_office_filing_cabinet",
+    "type": "item_group",
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for an office filing cabinet containing folders and documents.",
+    "subtype": "collection",
+    "entries": [
+      { "item": "file", "count": [ 1, 15 ], "prob": 50 },
+      { "item": "paper", "count": [ 20, 300 ], "prob": 60 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/office.json]:
```suggestion
      { "item": "paper", "count": [ 2, 30 ], "prob": 60 },
```
Each count of paper is 150 sheets, 300*150 for 45 000 sheets is probably a bit much?

File: data/json/furniture_and_terrain/furniture-storage.json [hunk 4]
```diff
@@ -433,6 +433,33 @@
...
+    "id": "f_clothing_rail",
+    "name": "clothing rail",
+    "description": "A rail for hanging clothes on.",
+    "symbol": "{",
+    "color": "light_gray",
+    "move_cost_mod": -1,
+    "coverage": 10,
+    "required_str": 8,
+    "flags": [ "TRANSPARENT", "FLAMMABLE_HARD", "PLACE_ITEM", "BLOCKSDOOR" ],
+    "deconstruct": { "items": [ { "item": "pipe", "count": [ 4, 6 ] }, { "item": "sheet_metal_small", "count": [ 0, 4 ] } ] }, { "item": "wheel_casters", "count": 1 }
```
Comment (I-am-Erk) [inline data/json/furniture_and_terrain/furniture-storage.json]:
```suggestion
      {
    "deconstruct": {
      "items": [
        { "item": "pipe", "count": [ 4, 6 ] },
        { "item": "sheet_metal_small", "count": [ 0, 4 ] },
        { "item": "wheel_casters", "count": 1 }
      ]
    }
  }
```

Will need some linting after this

File: data/json/itemgroups/SUS/clothes_store.json [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    "subtype": "collection",
+    "entries": [
+      {
+        "distribution": [
+          {
+            "collection": [
+              { "item": "awl_steel", "prob": 70 },
+              { "item": "needle_curved", "prob": 75 },
+              { "item": "scissors", "prob": 90 },
+              { "item": "knitting_needles", "prob": 70 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/clothes_store.json]:
```suggestion
              { "item": "knitting_needles", "prob": 20 },
```
Knitting needles are pretty unlikely in a tailor drawer I think

File: data/json/itemgroups/SUS/office.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    "subtype": "collection",
+    "entries": [
+      { "item": "eink_tablet_pc", "prob": 20 },
+      { "item": "laptop", "prob": 60 },
+      { "item": "usb_drive", "prob": 20 },
+      { "item": "mobile_memory_card_used", "count": [ 1, 4 ], "prob": 15 },
+      { "item": "mobile_memory_card_encrypted", "count": [ 1, 2 ], "prob": 15 },
+      { "item": "file", "count": [ 1, 8 ], "prob": 50 },
+      { "item": "paper", "count": [ 2, 9 ], "prob": 60 },
+      { "item": "pen", "prob": 95 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/office.json]:
```suggestion
      { "item": "pen", "count": [ 1, 10 ], "prob": 95 },
```

File: data/json/itemgroups/SUS/office.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    ]
+  },
+  {
+    "id": "SUS_office_filing_cabinet",
+    "type": "item_group",
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for an office filing cabinet containing folders and documents.",
+    "subtype": "collection",
+    "entries": [
+      { "item": "file", "count": [ 1, 15 ], "prob": 50 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/office.json]:
```suggestion
      { "item": "file", "count": [ 1, 15 ], "prob": 90 },
```

File: data/json/itemgroups/SUS/office.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      { "item": "paper", "count": [ 2, 30 ], "prob": 60 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/office.json]:
```suggestion
      { "item": "paper", "count": [ 2, 30 ], "prob": 80 },
```

File: data/json/itemgroups/SUS/clothes_store.json [hunk 9]
```diff
@@ -0,0 +1,136 @@
+[
+  {
+    "id": "SUS_mannequin_outdoor",
+    "type": "item_group",
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for a mannequin displaying outdoor gear.",
+    "subtype": "collection",
+    "entries": [ { "group": "clothing_outdoor_set" } ]
```
[the remark this answers is not in the document]
Comment (ymber) [reply data/json/itemgroups/SUS/clothes_store.json]: The outdoor set itemgroups have an appropriate set of stuff that's not specifically outdoor gear.

File: data/json/itemgroups/SUS/clothes_store.json [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    "subtype": "collection",
+    "entries": [ { "group": "dresses_womens" } ]
+  },
+  {
+    "id": "SUS_mannequin_formal_mens",
+    "type": "item_group",
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for a mannequin displaying formal clothes.",
+    "subtype": "collection",
+    "entries": [ { "group": "formal_set_mens" } ]
```
[the remark this answers is not in the document]
Comment (ymber) [reply data/json/itemgroups/SUS/clothes_store.json]: Shoes are in `formal_set_mens`.

File: data/json/itemgroups/SUS/clothes_store.json [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+    "id": "SUS_tailoring_bookcase",
+    "type": "item_group",
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for a set of tailoring manuals and references.",
+    "subtype": "collection",
+    "entries": [
+      { "item": "mag_tailor", "count": [ 1, 10 ], "prob": 80 },
+      { "item": "manual_tailor", "count": [ 1, 4 ], "prob": 80 },
+      { "item": "tailor_portfolio", "prob": 20 },
+      { "item": "tailor_japanese", "prob": 10 },
```
[the remark this answers is not in the document]
Comment (ymber) [reply data/json/itemgroups/SUS/clothes_store.json]: I think it's reasonable to allow both to appear. A well stocked tailors library is fairly likely to have stuff from around the world.

File: data/json/itemgroups/SUS/clothes_store.json [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    "//": "SUS item groups are collections that contain a reasonable realistic distribution of items that might spawn in a given storage furniture.",
+    "//2": "This group is for a container of fasteners for making clothing.",
+    "subtype": "collection",
+    "entries": [
+      { "item": "button_plastic", "count": [ 1, 5 ], "prob": 75 },
+      { "item": "button_steel", "count": [ 1, 5 ], "prob": 75 },
+      { "item": "button_wood", "count": [ 1, 5 ], "prob": 75 },
+      { "item": "snapfastener_steel", "count": [ 1, 5 ], "prob": 75 },
+      { "item": "zipper_long_plastic", "count": [ 1, 8 ], "prob": 75 },
+      { "item": "zipper_short_plastic", "count": [ 1, 8 ], "prob": 75 }
```
[the remark this answers is not in the document]
Comment (ymber) [reply data/json/itemgroups/SUS/clothes_store.json]: This feels like a crude solution. Do we need finer control of spawn quantity probabilities in these groups?

**Pull request merged**